examples/spline_example.cpp

```diff
@@ -0,0 +1,291 @@
+{
+  // trajectory_state = state;
+  trajectory_running = false;
+  std::string report = "?";
+  switch (state)
+  {
+    case control::TrajectoryResult::TRAJECTORY_RESULT_SUCCESS:
+      report = "success";
+      break;
+    case control::TrajectoryResult::TRAJECTORY_RESULT_CANCELED:
+      report = "canceled";
+      break;
+    case control::TrajectoryResult::TRAJECTORY_RESULT_FAILURE:
+    default:
+      report = "failure";
+      break;
+  }
+  std::cout << "\033[1;32mTrajectory report: " << report << "\033[0m\n" << std::endl;
+}
+
+int main(int argc, char* argv[])
+{
+  urcl::setLogLevel(urcl::LogLevel::INFO);
+
+  // Parse the ip arguments if given
+  std::string robot_ip = DEFAULT_ROBOT_IP;
+  if (argc > 1)
+  {
+    robot_ip = std::string(argv[1]);
+  }
+
+  // Making the robot ready for the program by:
+  // Connect the the robot Dashboard
+  g_my_dashboard.reset(new DashboardClient(robot_ip));
+  if (!g_my_dashboard->connect())
+  {
+    URCL_LOG_ERROR("Could not connect to dashboard");
+    return 1;
+  }
+
+  // Stop program, if there is one running
+  if (!g_my_dashboard->commandStop())
+  {
+    URCL_LOG_ERROR("Could not send stop program command");
+    return 1;
+  }
+
+  // Power it off
+  if (!g_my_dashboard->commandPowerOff())
+  {
+    URCL_LOG_ERROR("Could not send Power off command");
+    return 1;
+  }
+
+  // Power it on
+  if (!g_my_dashboard->commandPowerOn())
+  {
+    URCL_LOG_ERROR("Could not send Power on command");
+    return 1;
+  }
+
+  // Release the brakes
+  if (!g_my_dashboard->commandBrakeRelease())
+  {
+    URCL_LOG_ERROR("Could not send BrakeRelease command");
+    return 1;
+  }
+
+  // Now the robot is ready to receive a program
+
+  std::unique_ptr<ToolCommSetup> tool_comm_setup;
+  const bool HEADLESS = true;
+  g_my_driver.reset(new UrDriver(robot_ip, SCRIPT_FILE, OUTPUT_RECIPE, INPUT_RECIPE, &handleRobotProgramState, HEADLESS,
+                                 std::move(tool_comm_setup), CALIBRATION_CHECKSUM));
+
+  g_my_driver->registerTrajectoryDoneCallback(&handleTrajectoryState);
+
+  // Once RTDE communication is started, we have to make sure to read from the interface buffer, as
+  // otherwise we will get pipeline overflows. Therefore, do this directly before starting your main
+  // loop.
+
+  g_my_driver->startRTDECommunication();
+
+  std::unique_ptr<rtde_interface::DataPackage> data_pkg = g_my_driver->getDataPackage();
+
+  if (data_pkg)
+  {
+    // Read current joint positions from robot data
+    if (!data_pkg->getData("actual_q", g_joint_positions))
+    {
+      // This throwing should never happen unless misconfigured
+      std::string error_msg = "Did not find 'actual_q' in data sent from robot. This should not happen!";
+      throw std::runtime_error(error_msg);
+    }
+  }
+
+  const unsigned number_of_points = 5;
+  const double s_pos[number_of_points] = { 4.15364583e-03, 4.15364583e-03, -1.74088542e-02, 1.44817708e-02,
```

Comment:
There seems to be an error in unit conversion. The motions done by the robot are in fact a lot larger than this:
```
0: 0.0331154
1: 0.0331154
2: 0.0115529
3: 0.0434435
4: 0.0434435
```